test/session.test.ts

```diff
@@ -229,6 +229,31 @@ describe("Session", () => {
     return expect((await headers).get("X-Test-Header")).toEqual("test");
   });
 
+  it("Should allow api option header based on ensureSerializableResponse", async () => {
+    const headers = new Promise<Headers>((resolve) => {
+      server.use(
+        http.post(
+          "http://ftrack.test/api",
+          (info) => {
+            resolve(info.request.headers as any);
+            return HttpResponse.json(getInitialSessionQuery());
+          },
+          { once: true },
+        ),
+      );
+    });
+
+    new Session(
+      credentials.serverUrl,
+      credentials.apiUser,
+      credentials.apiKey,
+      {
+        ensureSerializableResponse: false,
+      },
+    );
+    return expect((await headers).get("ftrack-api-options")).toBeFalsy();
```

Comment:
i think this test would have passed with the previous implementation as well. check to see if you can use toHaveProperty instead?